nvflare/app_opt/xgboost/executor_base.py

```diff
@@ -81,12 +73,21 @@ def __init__(
         self._ca_cert_path = None
         self._client_key_path = None
         self._client_cert_path = None
+        self._server_address = "localhost"
 
     @abstractmethod
     def xgb_train(self, params: XGBoostParams, fl_ctx: FLContext) -> Shareable:
         """Main XGBoost training method"""
         pass
 
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.START_RUN:
+            engine = fl_ctx.get_engine()
+            sp = engine.client.overseer_agent.get_primary_sp()
```

Comment:
Is overseer_agent required in non-HA mode?

Reply:
It should be there for POC mode, let me check for simulator mode